ui/js/controllers/bugfiler.js

```diff
@@ -0,0 +1,224 @@
+                    }
+                }
+                if(i !== 0) {
+                    thisFailure += "\n";
+                }
+                thisFailure += allFailures[i].join(" | ");
+                $("#modalFailureList");
+            }
+            $("#modalFailureList").val(thisFailure);
+
+            $scope.findProduct();
+        };
+
+        $uibModalInstance.parsedSummary = "";
+        $uibModalInstance.initiate = $scope.initiate;
+        $uibModalInstance.possibleFilename = "";
+
+        /*
+         *  Remove extraneous junk from the start of the summary line
+         *  and try to find the failing test name from what's left
+         */
+        $uibModalInstance.parseSummary = function(summary) {
+            var omittedLeads = ["TEST-UNEXPECTED-FAIL", "PROCESS-CRASH", "TEST-UNEXPECTED-ERROR", "TEST-UNEXPECTED-TIMEOUT"];
+            summary = summary.split(" | ");
+
+            for(var i=0; i < omittedLeads.length; i++) {
+                if(summary[0].search(omittedLeads[i]) >= 0) {
+                    summary.shift();
+                }
+            }
+
+            $uibModalInstance.possibleFilename = summary[0].split("/").pop();
+
+            return [summary, $uibModalInstance.possibleFilename];
+        };
+
+        $scope.toggleFilerSummaryVisibility = function() {
+            $scope.isFilerSummaryVisible = !$scope.isFilerSummaryVisible;
+        };
+
+        $scope.isFilerSummaryVisible = false;
+
+        /*
+         *  Attempt to find a good product/component for this failure
+         */
+        $scope.findProduct = function() {
+            $scope.suggestedProducts = [];
+            var failurePath = $uibModalInstance.parsedSummary[0][0];
+            var failurePathRoot = failurePath.split("/")[0];
+
+            // Look up the product via the root of the failure's file path
+            if(thBugzillaProductObject[failurePathRoot]) {
+                $scope.suggestedProducts.push(thBugzillaProductObject[failurePathRoot][0]);
+            }
+
+            // Look up product suggestions via Bugzilla's api
+            var productSearch = $scope.productSearch;
+
+            if(productSearch) {
+                $http.get("https://bugzilla.mozilla.org/rest/prod_comp_search/" + productSearch + "?limit=5").then(function(request) {
+                    var data = request.data;
+                    $scope.suggestedProducts = [];
+                    for(var i = 0; i < data.products.length;i++) {
+                        if(data.products[i].product && data.products[i].component) {
+                            $scope.suggestedProducts.push(data.products[i].product + " :: " + data.products[i].component);
+                        }
+                    }
+                });
+            }
+        };
+
+        /*
+         *  This is called once intermittent.html's ng-repeat is finished to select the first product listed
+         */
+        $scope.focusProduct = function() {
```

Comment:
This is a spot where angular can make your life LOTS easier.  We actually don't need this function at all.  Here's where you want to assign a value to the scope that keeps track of which suggested product is currently selected.  You can set it and get it just by looking at ``$scope.selectedProduct``.  I'll put comments in the spots where the changes are needed, but here's a summary:

After you get the results of the product search in ``findProduct``, you loop through the results and build your ``suggestedProducts``.  After that loop, add a line:

     $scope.selectedProduct = $scope.suggestedProducts[0];

Then, in the HTML for the ``ng-repeat`` for these options, take out the ``ng-init`` and for each ``<input>`` element, add in ``ng-model="selectedProduct"``

then down in ``submitFilter`` you can just get the value of ``$scope.selectedProduct`` and split it or give an error if it's not set or what have you.